x/bor/side_handler.go

```diff
@@ -0,0 +1,153 @@
+package bor
+
+import (
+	"bytes"
+	"fmt"
+	"strconv"
+
+	tmTypes "github.com/tendermint/tendermint/types"
+
+	hmCommon "github.com/maticnetwork/heimdall/common"
+
+	"github.com/maticnetwork/heimdall/types/common"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/maticnetwork/heimdall/helper"
+	hmTypes "github.com/maticnetwork/heimdall/types"
+	"github.com/maticnetwork/heimdall/x/bor/keeper"
+	"github.com/maticnetwork/heimdall/x/bor/types"
+	abci "github.com/tendermint/tendermint/abci/types"
+	tmprototypes "github.com/tendermint/tendermint/proto/tendermint/types"
+)
+
+// NewSideTxHandler returns a side handler for "span" type messages.
+func NewSideTxHandler(k keeper.Keeper, contractCaller helper.IContractCaller) hmTypes.SideTxHandler {
+	return func(ctx sdk.Context, msg sdk.Msg) abci.ResponseDeliverSideTx {
+		ctx = ctx.WithEventManager(sdk.NewEventManager())
+		switch msg := msg.(type) {
+		case *types.MsgProposeSpan:
+			return SideHandleMsgSpan(ctx, k, *msg, contractCaller)
+		default:
+			errMsg := fmt.Sprintf("unrecognized %s message type: %T", types.ModuleName, msg)
+			fmt.Println(errMsg)
+			return abci.ResponseDeliverSideTx{
+				Code: uint32(6), // TODO should be changed like `sdk.CodeUnknownRequest `
```

Comment:
Use `Code: sdkerrors.ErrUnknownRequest.ABCICode(),` instead.
Can you also change this in staking, clerk and checkpoint?

Reply:
ok, other modules we'll take it up later.